unofficial/c511027102.lua

```diff
@@ -0,0 +1,102 @@
+--ZW－荒鷲激神爪
+--ZW - Eagle Claw (anime)
+--Script by Rundas
+local s,id=GetID()
+function s.initial_effect(c)
+	--normal summon without tribute
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetCode(EFFECT_SUMMON_PROC)
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCondition(s.sumcon)
+	c:RegisterEffect(e1)
+	--equip
+	local e2=Effect.CreateEffect(c)
+	e2:SetCategory(CATEGORY_EQUIP)
+	e2:SetDescription(aux.Stringid(id,1))
+	e2:SetType(EFFECT_TYPE_IGNITION)
+	e2:SetProperty(EFFECT_FLAG_CARD_TARGET)
+	e2:SetRange(LOCATION_MZONE)
+	e2:SetTarget(s.eqtg)
+	e2:SetOperation(s.eqop)
+	c:RegisterEffect(e2)
+	--negate
+	local e3=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,2))
+	e3:SetCategory(CATEGORY_DISABLE)
+	e3:SetType(EFFECT_TYPE_QUICK_O)
+	e3:SetCode(EVENT_CHAINING)
+	e3:SetRange(LOCATION_SZONE)
+	e3:SetCondition(s.negcon)
+	e3:SetCost(s.negcost)
+	e3:SetOperation(s.negop)
+	c:RegisterEffect(e3)
+end
+
+s.listed_names={56840427}
+
+function s.sumcon(e,c)
+	if c==nil then return true end
+	local tp=c:GetControler()
+	return Duel.GetLP(tp)<=Duel.GetLP(1-tp)-2000
+		and Duel.GetLocationCount(tp,LOCATION_MZONE)>0
+end
+
+function s.filter(c)
+	return c:IsFaceup() and c:IsCode(56840427)
+end
+
+function s.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and s.filter(chkc) end
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
+		and Duel.IsExistingTarget(s.filter,tp,LOCATION_MZONE,0,1,nil) end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_EQUIP)
+	Duel.SelectTarget(tp,s.filter,tp,LOCATION_MZONE,0,1,1,nil)
+end
+
+function s.eqop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
+	if not c:IsRelateToEffect(e) or c:IsFacedown() then return end
+	local tc=Duel.GetFirstTarget()
+	if Duel.GetLocationCount(tp,LOCATION_SZONE)<=0 or not tc or tc:GetControler()~=tp or tc:IsFacedown() or not tc:IsRelateToEffect(e) then
+		Duel.SendtoGrave(c,REASON_EFFECT)
+		return
+	end
+	Duel.Equip(tp,c,tc,true)
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetCode(EFFECT_EQUIP_LIMIT)
+	e1:SetReset(RESET_EVENT+RESETS_STANDARD)
+	e1:SetValue(s.eqlimit)
+	e1:SetLabelObject(tc)
+	c:RegisterEffect(e1)
+	--atkup
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_EQUIP)
+	e2:SetCode(EFFECT_UPDATE_ATTACK)
+	e2:SetValue(2000)
+	e2:SetReset(RESET_EVENT+RESETS_STANDARD)
+	c:RegisterEffect(e2)
+end
+
+function s.eqlimit(e,c)
+	return c==e:GetLabelObject()
+end
+
+function s.negcon(e,tp,eg,ep,ev,re,r,rp)
+	local ec=e:GetHandler():GetEquipTarget()
+	return ec and (ec==Duel.GetAttacker() or ec==Duel.GetAttackTarget()) and ec:GetBattleTarget() and rp~=tp and Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_SZONE

```

Comment:
You don't need 
```lua
Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_SZONE
``` 
since unlike OCG, it doesn't only negate Trap in S/T Zone. 
And you need to check 
```lua
re:IsHasType(EFFECT_TYPE_ACTIVATE)
``` 
as you only negate Trap Card's activation, not their effects. 
Also, it's preferable to check whether ec:GetBattleTarget() is controlled by opponent in case of redirected attacks.